File: solutions/_1461_check_if_a_string_contains_all_binary_codes_of_size_k/solution.cpp

#include <iostream>
#include <string>
#include <set>
using namespace std;

class Solution
{
public:
    bool hasAllCodes(string s, int k)
    {
        if (k > s.size())
            return false;

        set<string> seen;
        for (int i = 0; i <= s.size() - k; i++)
        {
            seen.insert(s.substr(i, k));
        }

        return seen.size() == 1 << k;
    }
};

/*
https://leetcode.com/problems/check-if-a-string-contains-all-binary-codes-of-size-k/
*/

File: solutions/_1461_check_if_a_string_contains_all_binary_codes_of_size_k/solution.py
class Solution:
    def hasAllCodes(self, s: str, k: int) -> bool:
        return len(set(s[i : i + k] for i in range(len(s) - k + 1))) == 2**k


# https://leetcode.com/problems/check-if-a-string-contains-all-binary-codes-of-size-k/
